platform/mds/mds-web/src/main/resources/webapp/js/controllers.js

```diff
@@ -3743,13 +3747,34 @@
         */
         $scope.addEntityInstance = function () {
             blockUI();
-
-            var values = $scope.currentRecord.fields;
+            $scope.addEntityInstanceBase($scope.currentRecord);
+        };
+        $scope.addEntityInstanceBase = function (inst)
+        {
+            var values = inst.fields;
             angular.forEach (values, function(value, key) {
                 value.value = value.value === 'null' ? null : value.value;
             });
+            for(var i in inst.fields) {
+                var fd = inst.fields[i];
+
+                // {readonly, uiFilterable, . . .} = fields(FieldDto) - fields(FieldRecord)
+                delete fd["readOnly"];
```

Comment:
Extra fields crash the json mapper on the server side.  So for each field I either had to add the member, or delete it before it returns to the server.   This comments indicates the criterion for delete:

// {readonly, uiFilterable, . . .} = fields(FieldDto) - fields(FieldRecord)

What is the charter of FieldRecord?  Is there a good reason for FieldRecord to different from FieldDto?